breez-sdk-wrapper/src/mock.rs

```diff
@@ -0,0 +1,552 @@
+            .current_timestamp()
+            .min_final_cltv_expiry_delta(144)
+            .build_signed(|hash| Secp256k1::new().sign_ecdsa_recoverable(hash, &private_key))
+            .unwrap();
+
+        let description = Option::from(req.description);
+
+        if let PaymentSuccess::Success = &*PAYMENT_SUCCESS.lock().unwrap() {
+            *LN_BALANCE_MSAT.lock().unwrap() += req.amount_msat;
+            PAYMENTS.lock().unwrap().push(Payment {
+                id: Utc::now().timestamp().to_string(), // Placeholder. Id is probably never used
+                payment_type: PaymentType::Received,
+                payment_time: Utc::now().timestamp(),
+                amount_msat: req.amount_msat,
+                fee_msat: lsp_fee.unwrap_or(0),
+                status: PaymentStatus::Complete,
+                description: description.clone(),
+                details: PaymentDetails::Ln {
+                    data: LnPaymentDetails {
+                        payment_hash: payment_hash.to_string(),
+                        label: "".to_string(),
+                        destination_pubkey: NODE_PUBKEY.to_string(),
+                        payment_preimage: preimage,
+                        keysend: false,
+                        bolt11: invoice.to_string(),
+                        lnurl_success_action: None,
+                        ln_address: None,
+                        lnurl_metadata: None,
+                        lnurl_withdraw_endpoint: None,
+                        swap_info: None,
+                    },
+                },
+            });
+        }
+
+        Ok(ReceivePaymentResponse {
+            ln_invoice: LNInvoice {
+                bolt11: invoice.to_string(),
+                network: Network::Bitcoin,
+                payee_pubkey: NODE_PUBKEY.to_string(),
+                payment_hash: SAMPLE_PAYMENT_HASH.to_string(),
+                description,
+                description_hash: None,
+                amount_msat: Some(req.amount_msat),
+                timestamp: Utc::now().timestamp() as u64,
+                expiry: 0,
+                routing_hints: vec![],
+                payment_secret: Vec::from(SAMPLE_PAYMENT_SECRET.as_bytes()),
+                min_final_cltv_expiry_delta: 144,
+            },
+            opening_fee_params: None,
+            opening_fee_msat: lsp_fee,
+        })
+    }
+
+    pub async fn service_health_check(&self) -> SdkResult<ServiceHealthCheckResponse> {
+        Ok(ServiceHealthCheckResponse {
+            status: HEALTH_STATUS.lock().unwrap().clone(),
+        })
+    }
+
+    // Not useful for the mock, but required to keep same interface
+    pub async fn report_issue(&self, _req: ReportIssueRequest) -> SdkResult<()> {
+        Ok(())
+    }
+
+    pub fn node_info(&self) -> SdkResult<NodeState> {
+        let balance = LN_BALANCE_MSAT.lock().unwrap().clone();
+
+        Ok(NodeState {
+            id: NODE_PUBKEY.to_string(),
```

Comment:
Well, that was kind of historic, at first provided a hard-coded Invoice rather than a generated one, and I wanted the node public key to match that invoice. But of course, that's not necessary anymore.

What do you mean by 'the one provided by 3L' though? 3L doesn't provide a key pair, but just a seed right? So certainly, I could derive a key from the seed, and I'd probably go ahead and implement that, but there's a catch:

The way Core-Lightning derives the node key pair [does not follow BIP32 and is non-trivial](https://github.com/ElementsProject/lightning/issues/3687). To keep things easy, I'd just derive _some_ key pair using _some_ derivation path, which would then result in the mock displaying a different node id using the same seed as the actual breez-sdk implementation does.

WDYT?

Reply:
@andrei-21 I've incorporated #901 and I like it much better. Thank you!

What I added to the change:
* Call the library `breez-sdk-mock` instead of `breez-sdk-wrapper`, now that it's not wrapping breez-sdk anymore
* I've removed `mock.rs` and instead moved the code into lib.rs itself

Here's the resulting commit: 87454460b4a6bd07cb52712140cce5517d9c3d71